### PR TITLE
New Settings-System

### PR DESCRIPTION
**Pull request by SIMULATAN**
# New Settings Base
### Checklist

- [X] I have checked the PRs for upcoming features/bug fixes.
- [X] I have read the [contributing guidelines](/CONTRIBUTING.md).
- [X] I agree with the [license](/LICENSE).

### Changes
- [ ] Refactoring
- [X] Internal code
- [ ] External Code / Libraries
- [ ] Documentation
- [ ] Other: \_____

Closes Issue: #168 

## Description
Replace the old reflections-based settings system with a new, OOP-Oriented one

### CODE REVIEW

**Comment by RealYusufIsmail** [comment]
This pr is blocked until #181 and #183 are merged

File: src/main/java/com/github/black0nion/blackonionbot/wrappers/jda/BlackGuild.java [hunk 1]
```diff
@@ -59,9 +75,11 @@ public static BlackGuild from(@Nullable final Guild guild) {
 		try {
 			return guilds.get(guild);
 		} catch (final Exception e) {
-			if (!(e instanceof IllegalStateException)) {
-				e.printStackTrace();
+			if (e.getCause() != null && (e instanceof ExecutionException || e instanceof UncheckedExecutionException)) {
+				e.getCause().printStackTrace();
```
Comment (RealYusufIsmail) [inline src/main/java/com/github/black0nion/blackonionbot/wrappers/jda/BlackGuild.java:79]: Throw a logger.error here

File: src/main/java/com/github/black0nion/blackonionbot/misc/GuildType.java [hunk 2]
```diff
@@ -1,5 +1,7 @@
 package com.github.black0nion.blackonionbot.misc;
 
+import com.github.black0nion.blackonionbot.systems.settings.Setting;
```
Comment (RealYusufIsmail) [inline src/main/java/com/github/black0nion/blackonionbot/misc/GuildType.java:3]: why has this import been added

File: src/main/java/com/github/black0nion/blackonionbot/systems/antiswear/AntiSwearSystem.java [hunk 3]
```diff
@@ -50,7 +50,7 @@ public static boolean check(final BlackGuild guild, final BlackMember author, fi
 		try {
 			if (messageContent.equalsIgnoreCase("")) return false;
 			// check for whitelist
-			final List<String> whitelist = guild.getAntiSwearWhitelist();
+			final List<String> whitelist = guild.getAntiSwearWhitelist().getValue();
```
Comment (RealYusufIsmail) [inline src/main/java/com/github/black0nion/blackonionbot/systems/antiswear/AntiSwearSystem.java:53]: can you not have it were getValued is called automatically instead having to always called e.g getAntiSwearWhitelist() calls getValue.
This applies to all the other getValues.

**Review by RealYusufIsmail — CHANGES_REQUESTED** [review]

File: src/main/java/com/github/black0nion/blackonionbot/systems/antiswear/AntiSwearSystem.java (hunk 3 above)
Comment (SIMULATAN) [reply src/main/java/com/github/black0nion/blackonionbot/systems/antiswear/AntiSwearSystem.java:53]: Already thought about that, but I'm using Lombok rn which doesn't support that, but I'll have a look at it

Comment (RealYusufIsmail) [reply src/main/java/com/github/black0nion/blackonionbot/systems/antiswear/AntiSwearSystem.java:53]: I would advise against using Lombok. It throws a compile error when using its ```@getters and @setter ``` .

File: build.gradle [hunk 4]
```diff
@@ -79,6 +79,8 @@ dependencies {
     implementation "io.prometheus:simpleclient_hotspot:0.15.0"
     implementation "io.prometheus:simpleclient_httpserver:0.15.0"
 
+    compileOnly "org.projectlombok:lombok:1.18.22"
+
```
Comment (RealYusufIsmail) [inline build.gradle:83]:
```suggestion
```

Do not use Lombok see previous message.

**Review by RealYusufIsmail — CHANGES_REQUESTED** [review]

**Comment by SIMULATAN** [comment]
Replaced by #320

**Pull request closed without merging**